### PR TITLE
[ENH] Preprocess: Add dropout-based gene selection

### PR DESCRIPTION
**Pull request by VesnaT**
##### Issue

Implements preprocessing part of #348

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/single_cell/widgets/owscpreprocess.py [hunk 1]
```diff
@@ -394,6 +394,43 @@ def __repr__(self):
         return text
 
 
+class DropoutEditor(ScBaseEditor):
+    DEFAULT_N_GENS = 1000
```
Comment (pavlin-policar) [inline orangecontrib/single_cell/widgets/owscpreprocess.py]: DEFAULT_N_GENES

File: orangecontrib/single_cell/preprocess/scpreprocess.py [hunk 2]
```diff
@@ -164,3 +165,65 @@ def _filter_columns(data, indices):
...
         domain = Domain(tuple(np.array(attrs)[indices]), cls, metas)
         return data.transform(domain)
+
+
+class DropoutGeneSelection(Preprocess):
+    def __init__(self, n_genes=None):
+        self.n_genes = n_genes
+
+    def __call__(self, data):
+        selected = self._get_selection_mask(data.X, n=self.n_genes)
```
Comment (pavlin-policar) [inline orangecontrib/single_cell/preprocess/scpreprocess.py]: We never talked about this, but the gene selection implementation, by default, removes genes that have "low detection". In Orange, this same functionality is achieved through the "Filter" widget, so this should probably be disabled here. Add `atleast=0` to the function call to remove the gene filtering.

**Comment by JakaKokosar** [comment]
@VesnaT I have enabled CI from Forks again (It was disabled for some reason). Next commit should trigger CI :)

**Comment by pavlin-policar** [comment]
@VesnaT Could you trigger a CI build so we can see the tests pass. Then I'll merge since this works as expected.

**Comment by pavlin-policar** [comment]
The reason the tests are failing is due to an old version of the Orange-Bioinformatics addon. I ran into the same issue yesterday on my local machine and pulling down master fixed the problem for me.

**Comment by pavlin-policar** [comment]
I will merge this since it works well on my local machine and half the tests pass. Where the tests fail, they are unrelated to the code here. I've seen them on my local machine before and this was a result of an old version of the bioinformatics addon.

**Pull request merged**